File: Cpp/Array/Sort 0s, 1s and 2s.cpp
//https://www.geeksforgeeks.org/problems/sort-an-array-of-0s-1s-and-2s4231/1
void sort012(int a[], int n)
{
    int i,j,k;
    i=0,k=n-1,j=0;
    while(j<=k){
        if(a[j]==0){
            a[j]=a[i];
            a[i]=0;
            i++;
            j++;
        }
        else if(a[j]==2){
            a[j]=a[k];
            a[k]=2;
            k--;
        }
        else if(a[j]==1){
            j++;
        }
    }
    
}

File: Python/Array/Sort 0s, 1s and 2s.py
//https://www.geeksforgeeks.org/problems/sort-an-array-of-0s-1s-and-2s4231/1
class Solution:
    # Function to sort an array of 0s, 1s, and 2s
    def sort012(self, arr):
        i, j, k = 0, 0, len(arr) - 1
        
        while j <= k:
            if arr[j] == 0:
                arr[i], arr[j] = arr[j], arr[i]
                i += 1
                j += 1
            elif arr[j] == 2:
                arr[j], arr[k] = arr[k], arr[j]
                k -= 1
            else:  # arr[j] == 1
                j += 1
